docs/design/908-proposal-process.md

```diff
@@ -0,0 +1,410 @@
+documented and the proposal will move to the appropriate state.
+
+[governance]: https://github.com/grafana/alloy/blob/main/GOVERNANCE.md
+[vote]: https://github.com/grafana/alloy/blob/main/GOVERNANCE.md#voting
+
+## Rationale
+
+Formalizing proposal creation and review ensures that all proposals are
+eventually given attention. Having proposals reviewed weekly ensures that
+proposals are kept moving forward and nobody is left waiting months for a
+decision to be made.
+
+To re-embrace the spirit of "design in the open," conversations about a
+proposal should be discoverable by new parties to make it easy to participate.
+Having a single source of truth for discussions (the proposal issue or design
+document PR) prevents concurrent conversations from fragmenting in multiple
+locations. This ensures that contributors know where to look for active
+discussions and important information doesn't get lost while trying to reach
+consensus.
+
+However, this process comes with some trade-offs, described in sections below.
+
+### Proposal lifecycle time
+
+Formalizing the proposal process ensures that all proposals are given
+attention. However, the process described here will slow down the acceptance
+rate of proposals.
+
+Because changing a proposal's state is tied to the frequency of proposal review
+process, a weekly proposal review process means that the acceptance cycle of
+most proposals is increased to take at least two or three weeks; up to one week
+to move to _Active_, at least one week to move to _Likely Accept_, and at least
+one week to move to _Accepted_. This does not include the time a proposal may
+spend in _Hold_.
+
+Three weeks is half the time of our release cadence, and so proposals opened
+too late in the cycle may need to wait two releases before being available to
+users.
+
+However, if proposals are needed more urgently, the proposal review group can
+fast-track proposals. This should mitigate the impact of this trade-off. If
+this mitigation is not sufficient, further proposals can be explored to reduce
+the time it takes to get a proposal approved and implemented into a release.
+
+### Conversational toil
+
+Using GitHub for all proposals ensures there is a single source of truth at a
+given time for discussion, and that all interested parties can participate in
+shaping the future of Alloy.
+
+However, GitHub is not well-suited for long-form discussions. This may lead to
+long threads that can be difficult to follow.
+
+There are some mitigation strategies to help with this:
+
+* Discussion on issues should be done conversationally rather than responding
+  to individual lines of text. This can help reduce the total number of
+  comments and make a conversation more digestible.
+
+* Authors of proposals can start with offline discussions and use alternative
+  tools (such as Google Docs) to prepare proposals for public presentation.
+
+* The proposal review group keeps discussions focused by preventing derailing
```

Comment:
Can you clarify how this will work? Say there are 4 different people mentioning 4 different concerns, will the review group mandate that we discuss them in some specific order, only one at a time?

Reply:
I don't expect the proposal review group to force conversations to happen in a specific order, especially as they only review proposals once a week.

But if someone brings up a concern that is not related to a proposal (such as "can we also have for loops" in a proposal for conditionals) or there are multiple related concerns that can be unified into a single theme (I don't have a concrete example for this unfortunately), the proposal review group may redirect things.

Reply:
It's not too important here, more of a nit, but I think that a task to keep discussions focused and preventing derailing is something that will be very hard to be effective at.